incopatible type with typescript v2.9.1

- [ ] I have searched the [issues](https://github.com/ant-design/ant-design/issues) of this repository and believe that this is not a duplicate.

### Version
3.5.4

### Environment
bowser 3.5.4

### Reproduction link
[![Edit on CodeSandbox](https://codesandbox.io/static/img/play-codesandbox.svg)](https://jnw46698m3.codesandbox.io/)

### Steps to reproduce
npm start

### What is expected?
compile successfully

### What is actually happening?
Error: 
antd\lib\_util\type.d.ts
(8,63): error TS2344: Type 'keyof T' does not satisfy the constraint 'string'.
  Type 'string | number | symbol' is not assignable to
type 'string'.
    Type 'number' is not assignable to type 'string'.


<!-- generated by ant-design-issue-helper. DO NOT REMOVE -->

--- a/components/_util/type.ts
+++ b/components/_util/type.ts
@@ -1,3 +1,3 @@
 export type Diff<T extends string, U extends string> = ({ [P in T]: P } &
   { [P in U]: never } & { [x: string]: never })[T];
-export type Omit<T, K extends keyof T> = Pick<T, Diff<keyof T, K>>;
+export type Omit<T, K extends keyof T> = Pick<T, Exclude<keyof T, K>>;

--- a/components/_util/type.ts
+++ b/components/_util/type.ts
@@ -1,3 +1,1 @@
-export type Diff<T extends string, U extends string> = ({ [P in T]: P } &
-  { [P in U]: never } & { [x: string]: never })[T];
 export type Omit<T, K extends keyof T> = Pick<T, Exclude<keyof T, K>>;

